griptape/tasks/action_subtask.py

```diff
@@ -31,13 +31,18 @@ class ActionSubtask(PromptTask):
     action_name: Optional[str] = field(default=None, kw_only=True)
     action_activity: Optional[str] = field(default=None, kw_only=True)
     action_input: Optional[dict] = field(default=None, kw_only=True)
+    _input: Optional[TextArtifact] = field(default=None, init=False)
 
     _tool: Optional[BaseTool] = None
     _memory: Optional[ToolMemory] = None
 
     @property
     def input(self) -> TextArtifact:
-        return TextArtifact(self.input_template)
+        if self._input is not None:
```

Comment:
What's the purpose if this?

Reply:
Also, not sure it's ever a good idea to swap out values in `TextArtifact`s since they might have other stuff (like metadata and embeddings data) not match. If we want to "update" the value, it needs to be a method that generates a new object.

Reply:
Comments were pending 🙃.  Multiple calls to `input` generate Artifacts with new `id`s. Is this intentional?

Reply:
Ah, I see! I think there is a better way to fix it (e.g., move this into a real property) but I think for now it's better to rebuild an artifact every time than potentially have mismatched "internals." Could result in one of those silent bug scenarios that are hard to trace down.

Reply:
Reverted.